metadata-io/src/main/java/com/linkedin/metadata/timeline/differ/SchemaDiffer.java

```diff
@@ -135,25 +135,30 @@ private static ChangeEvent getDescriptionChange(SchemaField baseField, SchemaFie
   // TODO: This could use some cleanup, lots of repeated logic and tenuous conditionals
   private static List<ChangeEvent> computeDiffs(SchemaMetadata baseSchema, SchemaMetadata targetSchema,
       Urn datasetUrn, ChangeCategory changeCategory) {
-    boolean isOrdinalBasedSchema = isSchemaOrdinalBased(targetSchema);
-    if (!isOrdinalBasedSchema) {
-      // Sort the fields by their field path.
-      if (baseSchema != null) {
-        sortFieldsByPath(baseSchema);
-      }
-      sortFieldsByPath(targetSchema);
+    // Sort the fields by their field path.
+    if (baseSchema != null) {
+      sortFieldsByPath(baseSchema);
     }
+    sortFieldsByPath(targetSchema);
 
     // Performs ordinal based diff, primarily based on fixed field ordinals and their types.
     SchemaFieldArray baseFields = (baseSchema != null ? baseSchema.getFields() : new SchemaFieldArray());
     SchemaFieldArray targetFields = targetSchema.getFields();
     int baseFieldIdx = 0;
     int targetFieldIdx = 0;
     List<ChangeEvent> changeEvents = new ArrayList<>();
+    Set<SchemaField> renamedFields = new HashSet<>();
     while (baseFieldIdx < baseFields.size() && targetFieldIdx < targetFields.size()) {
       SchemaField curBaseField = baseFields.get(baseFieldIdx);
       SchemaField curTargetField = targetFields.get(targetFieldIdx);
-      if (isOrdinalBasedSchema) {
+      //TODO: Re-evaluate ordinal processing?
+      int comparison = curBaseField.getFieldPath().compareTo(curTargetField.getFieldPath());
+      if (renamedFields.contains(curBaseField)) {
```

Comment:
This if block not super intuitive to read. Can you drop some comments here?

Reply:
Each section has a comment header addressing what the case is covering (which I already don't really love, I'm a big proponent of clean code and prefer not to use comments to address readability issues where possible), I can look into breaking it down further into submethods with descriptive names, but I did a lot of that with this PR already. Each section is basically a couple method calls and incrementing field counters now. The diff view is pretty hard to read though, opening it in the actual file view should be quite a bit easier to digest. I can pull out the ChangeEvent creation that's still in there into a separate method to reduce the bloat some more.